lib/active_merchant/billing/gateways/pay_junction_v2.rb

```diff
@@ -0,0 +1,224 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PayJunctionV2Gateway < Gateway
+      self.display_name = "PayJunction"
+      self.homepage_url = "https://www.payjunction.com/"
+
+      self.test_url = "https://api.payjunctionlabs.com/transactions"
+      self.live_url = "https://api.payjunction.com/transactions"
+
+      self.supported_countries = ["US"]
+      self.default_currency = "USD"
+      self.money_format = :dollars
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+
+      def initialize(options={})
+        requires!(options, :api_login, :api_password, :api_key)
+        super
+      end
+
+      def purchase(amount, payment_method, options={})
+        post = {}
+        add_invoice(post, amount, options)
+        add_payment_method(post, payment_method)
+
+        commit("purchase", post)
+      end
+
+      def authorize(amount, payment_method, options={})
+        post = {}
+        post[:status] = "HOLD"
+        add_invoice(post, amount, options)
+        add_payment_method(post, payment_method)
+
+        commit("authorize", post)
+      end
+
+      def capture(amount, authorization, options={})
+        post = {}
+        post[:status] = "CAPTURE"
+        post[:transactionId] = authorization
+        add_invoice(post, amount, options) if amount
+
+        commit("capture", post)
+      end
+
+      def void(authorization, options={})
+        post = {}
+        post[:status] = "VOID"
+        post[:transactionId] = authorization
+
+        commit("void", post)
+      end
+
+      def refund(money, authorization, options={})
```

Comment:
@shasum wondering about this `refund` method since it seems a little more complicated than a typical implementation. Of course there could be something I'm unaware of considering I didn't work on the remote tests like you did. 😄 

Is it not possible for us to skip doing a GET on a transaction for the `amountBase` and instead just pass the `money` parameter to the request? So, a more pared down implementation like-

```ruby
def refund(money, authorization, options={})
  post = {}
  post[:transactionId] = authorization
  post[:amountBase] = sprintf("%.2f", amount(money))
  commit("refund", post)
end
```

I'm thinking that we should leave it up to the method invoker to pass in the `money` param to set the refund amount?